scripts/hud/synchronizer.js

```diff
@@ -0,0 +1,344 @@
+		this.gainRatioHistory.push((this.sampleWeight * (speed - this.lastSpeed)) / (idealSpeed - this.lastSpeed));
+		this.gainRatioHistory.shift();
+		const gainRatio = this.getBufferedSum(this.gainRatioHistory);
+		this.lastAngle = viewAngle;
+		this.lastSpeed = speed;
+
+		const colorTuple = this.mom_hud_synchro_color_enable
+			? this.getColorTuple(gainRatio, strafeRight * angleDelta > idealDelta)
+			: COLORS.NEUTRAL;
+		const color = `gradient(linear, 0% 0%, 0% 100%, from(${colorTuple[0]}), to(${colorTuple[1]}))`;
+		const altColor = 'rgba(0, 0, 0, 0.0)';
+		let flow;
+
+		switch (this.mom_hud_synchro_mode) {
+			case 1: // "Half-width throttle"
+				flow = strafeRight * (this.mom_hud_synchro_flip_enable === 1 ? -1 : 1);
+				this.panels.container.style.flowChildren = flow < 0 ? 'left' : 'right';
+
+				const deltaRatio = (strafeRight * angleDelta) / idealDelta;
+				this.addToBuffer(this.strafeRatioHistory, this.sampleWeight * deltaRatio);
+				const strafeRatio = (0.5 * this.getBufferedSum(this.strafeRatioHistory)).toFixed(3);
+
+				this.panels.segments.forEach((segment) => {
+					segment.style.backgroundColor = altColor;
+				});
+
+				this.panels.segments[0].style.backgroundColor = color;
+				this.panels.segments[0].style.width = strafeRatio * 100 + '%';
+
+				// show needle
+				this.panels.needle.style.visibility = 'visible';
+				break;
+			case 2: // "Full-width throttle"
+				const absRatio = Math.abs(gainRatio).toFixed(3);
+				flow = direction * (this.mom_hud_synchro_flip_enable === 1 ? -1 : 1);
+				this.panels.container.style.flowChildren = flow < 0 ? 'left' : 'right';
+
+				this.panels.segments.forEach((segment) => (segment.style.backgroundColor = altColor));
+
+				this.panels.segments[0].style.backgroundColor = color;
+				this.panels.segments[0].style.width = absRatio * 100 + '%';
+
+				// hide needle
+				this.panels.needle.style.visibility = 'collapse';
+				break;
+			case 3: // "Strafe indicator"
+				this.panels.container.style.flowChildren = 'right';
+				const width = 10; // this value shows ~90% gain or better when bar touches needle
+				const offset =
+					this.sampleWeight *
+					Math.min(
+						Math.max(
+							0.5 - ((0.5 * syncDelta) / idealDelta) * (this.mom_hud_synchro_flip_enable === 1 ? -1 : 1),
+							0
+						),
+						1
+					);
+				this.addToBuffer(this.offsetHistory, offset);
+				this.panels.segments[0].style.width =
+					(this.getBufferedSum(this.offsetHistory) * (100 - width)).toFixed(3) + '%';
+				this.panels.segments[1].style.width = width + '%';
+				this.panels.segments[2].style.width = 50 + '%';
+				this.panels.segments[3].style.width = 50 + '%';
+				this.panels.segments.forEach(
+					(segment, i) => (segment.style.backgroundColor = i !== 1 ? altColor : color)
+				);
+
+				// show needle
+				this.panels.needle.style.visibility = 'visible';
+				break;
+			case 4: // "Synchronizer"
+				this.panels.container.style.flowChildren = 'right';
```

Comment:
Do these style changes need to happen every `onUpdate`? Shouldn't this happen only on convar change?

Reply:
style that doesn't change every frame is now handled in `setSynchroMode`